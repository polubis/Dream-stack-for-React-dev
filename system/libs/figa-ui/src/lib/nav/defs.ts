import type { ReactNode } from 'react';

interface NavProps {
  className?: string;
  children: ReactNode;
  actions: ReactNode;
  logo: ReactNode;
}

export type { NavProps };
