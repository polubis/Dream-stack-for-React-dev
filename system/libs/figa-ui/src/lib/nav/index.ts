export * from './nav';
export * from './defs';
