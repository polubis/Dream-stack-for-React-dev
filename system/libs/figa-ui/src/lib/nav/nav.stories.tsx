import type { Story, Meta } from '@storybook/react';

import { Nav, type NavProps } from '.';
import { Logo } from '../logo';
import { Button } from '../button';

export default {
  component: Nav,
  title: 'Nav',
} as Meta;

const Template: Story<NavProps> = () => {
  return (
    <Nav
      logo={<Logo />}
      actions={
        <>
          <Button size={2}>Register</Button>
          <Button size={2}>Sign In</Button>
        </>
      }
    >
      <Nav.Link variant="h6">
        <a href="/test">Articles</a>
      </Nav.Link>
      <Nav.Link variant="h6">
        <a href="/">Authors</a>
      </Nav.Link>
      <Nav.Link variant="h6">
        <a href="/">Creator</a>
      </Nav.Link>
      <Nav.Link variant="h6">
        <a href="/">Support</a>
      </Nav.Link>
      <Nav.Divider />
      <Nav.Link variant="h6">
        <a href="/">Admin</a>
      </Nav.Link>
      <Nav.Link variant="h6">
        <a href="/">Yours</a>
      </Nav.Link>
    </Nav>
  );
};

export const Default = Template.bind({});
Default.args = {};
