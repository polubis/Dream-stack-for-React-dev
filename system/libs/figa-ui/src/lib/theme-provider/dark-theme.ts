import type { Theme } from './defs';
import { tokens } from './tokens';

const dark: Theme = {
  outline: {
    color: tokens.common.white,
  },
  switch: {
    bg: tokens.gray[50],
    color: tokens.common.white,
    active: {
      bg: tokens.green[200],
    },
  },
  scroll: {
    thumb: tokens.gray[600],
    track: tokens.gray[550],
    thumbHover: tokens.gray[650],
  },
  divider: {
    default: {
      bg: tokens.common.white,
    },
    primary: {
      bg: tokens.primary[100],
    },
  },
  radio: {
    borderColor: tokens.common.white,
    active: {
      bg: tokens.green[200],
      borderColor: tokens.green[200],
    },
  },
  checkbox: {
    borderColor: tokens.common.white,
    checked: {
      bg: tokens.primary[50],
      borderColor: tokens.primary[50],
      color: tokens.common.white,
    },
  },
  tabs: {
    filled: {
      bg: tokens.gray[500],
      active: {
        borderColor: tokens.green[50],
      },
    },
  },
  chips: {
    color: tokens.common.white,
    borderColor: tokens.common.white,
    active: {
      color: tokens.green[50],
      borderColor: tokens.green[50],
    },
  },
  badge: {
    outlined: {
      primary: {
        color: tokens.primary[50],
        borderColor: tokens.primary[50],
      },
      secondary: {
        color: tokens.primary[100],
        borderColor: tokens.primary[100],
      },
      ok: {
        color: tokens.green[100],
        borderColor: tokens.green[100],
      },
      casual: {
        color: tokens.gray[150],
        borderColor: tokens.gray[150],
      },
    },
    filled: {
      primary: {
        color: tokens.common.black,
        bg: tokens.primary[50],
      },
      secondary: {
        color: tokens.common.black,
        bg: tokens.primary[100],
      },
      ok: {
        color: tokens.common.black,
        bg: tokens.green[100],
      },
      casual: {
        color: tokens.common.black,
        bg: tokens.gray[150],
      },
    },
  },
  thumbnail: {
    bg: tokens.backdrop[100],
    color: tokens.common.white,
  },
  filePicker: {
    outline: {
      color: tokens.primary[50],
    },
    invalid: {
      color: tokens.red[150],
      outline: tokens.red[150],
    },
  },
  field: {
    invalid: {
      color: tokens.red[100],
    },
    hint: {
      color: tokens.common.white,
    },
  },
  font: {
    default: {
      color: tokens.common.white,
    },
    primary: {
      color: tokens.primary[100],
    },
  },
  body: {
    bg: tokens.common.black,
  },
  alert: {
    filled: {
      info: {
        bg: tokens.gray[150],
        color: tokens.common.black,
      },
      ok: {
        bg: tokens.green[150],
        color: tokens.common.black,
      },
      error: {
        bg: tokens.red[50],
        color: tokens.common.white,
      },
      warn: {
        bg: tokens.yellow[50],
        color: tokens.common.black,
      },
    },
    outlined: {
      info: {
        color: tokens.common.white,
      },
      ok: {
        color: tokens.green[150],
      },
      error: {
        color: tokens.red[150],
      },
      warn: {
        color: tokens.yellow[50],
      },
    },
  },
  modal: {
    bg: tokens.common.black,
    border: tokens.dark[50],
    backdrop: tokens.backdrop[50],
  },
  button: {
    filled: {
      primary: {
        color: tokens.common.black,
        bg: tokens.primary[50],
        outlineColor: tokens.common.white,
      },
      secondary: {
        color: tokens.common.black,
        bg: tokens.primary[100],
        outlineColor: tokens.common.white,
      },
      tertiary: {
        color: tokens.common.black,
        bg: tokens.common.white,
        outlineColor: tokens.common.white,
      },
    },
    outlined: {
      primary: {
        color: tokens.primary[50],
        bg: 'transparent',
        borderColor: tokens.primary[50],
        outlineColor: tokens.common.white,
      },
      secondary: {
        color: tokens.primary[100],
        bg: 'transparent',
        borderColor: tokens.primary[100],
        outlineColor: tokens.common.white,
      },
      tertiary: {
        color: tokens.common.white,
        bg: 'transparent',
        borderColor: tokens.common.white,
        outlineColor: tokens.common.white,
      },
    },
    ghost: {
      hoverBg: tokens.dark[250],
    },
  },
  select: {
    bg: tokens.dark[50],
    color: tokens.common.white,
    placeholderColor: tokens.gray[50],
    hoverBg: tokens.gray[100],
    optionSeparator: tokens.gray[100],
    optionActiveColor: tokens.secondary[50],
  },
  nav: {
    bg: tokens.dark[100],
    borderColor: tokens.gray[300],
    bgWithOpacity: tokens.dark[200],
  },
  logo: {
    text: {
      first: tokens.purple[100],
      second: tokens.common.white,
      last: tokens.secondary[200],
    },
    bg: {
      first: {
        from: tokens.red[50],
        to: tokens.primary[250],
      },
      second: {
        from: tokens.secondary[250],
        to: tokens.primary[250],
      },
      last: {
        from: tokens.blue[200],
        to: tokens.primary[250],
      },
    },
  },
  progressCircle: {
    bg: tokens.dark[50],
  },
  input: {
    filled: {
      bg: tokens.dark[50],
      color: tokens.common.white,
      placeholderColor: tokens.gray[50],
      fxColor: tokens.common.white,
    },
    outlined: {
      borderColor: tokens.common.white,
      color: tokens.common.white,
      placeholderColor: tokens.gray[50],
      fxColor: tokens.common.white,
    },
    empty: {
      color: tokens.common.white,
      placeholderColor: tokens.gray[50],
      fxColor: tokens.common.white,
    },
    invalid: tokens.red[150],
  },
  placeholder: {
    filled: {
      bg: tokens.gray[300],
    },
    outlined: {
      bg: tokens.common.black,
      borderColor: tokens.gray[300],
    },
  },
  box: {
    filled: {
      bg: tokens.gray[300],
    },
    outlined: {
      bg: tokens.common.black,
      borderColor: tokens.gray[300],
    },
  },
  footer: {
    bg: tokens.common.black,
    borderColor: tokens.gray[300],
  },
  code: {
    bg: tokens.dark[150],
    gutters: tokens.dark[150],
    selectionBg: '#4d4d4d',
    foldPlaceholder: '#4d4d4d',
    tags: [
      { tag: 'angleBracket', color: '#da6bad' },
      { tag: 'arithmeticOperator', color: '#fff' },
      { tag: 'bitwiseOperator', color: '#fff' },
      { tag: 'blockComment', color: '#6A9955' },
      { tag: 'bool', color: '#569cd6' },
      { tag: 'brace', color: '#ffd70f' },
      { tag: 'bracket', color: '#da6bad' },
      { tag: 'className', color: '#3abfb0' },
      { tag: 'comment', color: '#6A9955' },
      { tag: 'compareOperator', color: '#fff' },
      { tag: 'controlKeyword', color: '#eba0d9' },
      { tag: 'definition', color: '#4fc1ff' },
      { tag: 'definitionKeyword', color: '#3981c5' },
      { tag: 'definitionOperator', color: '#fff' },
      { tag: 'derefOperator', color: '#fff' },
      { tag: 'function', color: '#dcdc9f' },
      { tag: 'keyword', color: '#3981c5' },
      { tag: 'labelName', color: '#4fc1ff' },
      { tag: 'lineComment', color: '#6A9955' },
      { tag: 'literal', color: '#ef8c59' },
      { tag: 'logicOperator', color: '#fff' },
      { tag: 'moduleKeyword', color: '#eba0d9' },
      { tag: 'name', color: '#4fc1ff' },
      { tag: 'null', color: '#569cd6' },
      { tag: 'number', color: '#accea8' },
      { tag: 'operatorKeyword', color: '#3abfb0' },
      { tag: 'paren', color: '#da6bad' },
      { tag: 'typeName', color: '#3abfb0' },
    ],
  },
  codeBlock: {
    header: {
      bg: tokens.gray[350],
      dots: {
        first: {
          bg: tokens.green[50],
        },
        second: {
          bg: tokens.yellow[50],
        },
        third: {
          bg: tokens.blue[650],
        },
      },
    },
    content: {
      bg: tokens.dark[150],
    },
  },
  list: {
    marker: {
      bg: tokens.primary[50],
    },
  },
  blockquote: {
    bg: tokens.gray[450],
    borderColor: tokens.gray[300],
  },
  loader: {
    primary: tokens.common.white,
    secondary: tokens.primary[50],
  },
  avatars: {
    restCounter: {
      bg: tokens.primary[50],
      color: tokens.common.black,
    },
  },
  creatorLayout: {
    borderColor: tokens.gray[300],
    bg: tokens.common.black,
  },
};

export { dark };
