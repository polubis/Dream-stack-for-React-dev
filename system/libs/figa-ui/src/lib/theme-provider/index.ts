export * from './theme-provider';
export * from './defs';
