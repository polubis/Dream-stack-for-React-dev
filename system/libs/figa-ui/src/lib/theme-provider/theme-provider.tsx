import { useState, createContext, useMemo, useContext } from 'react';
import { ThemeProvider as StyledThemeProvider } from 'styled-components';
import type {
  ThemeKey,
  ThemeProviderProps,
  ThemeProviderValue,
  ThemesList,
  UnsafeThemeProviderValue,
  GetDefaultValue,
} from './defs';
import { themes } from './themes';
import { GlobalStyle } from './global-style';

const Ctx = createContext<UnsafeThemeProviderValue>(null);

const defaultKey = 'dark' as ThemeKey;

const getDefaultValue: GetDefaultValue = ({
  key = defaultKey,
  // eslint-disable-next-line @typescript-eslint/no-empty-function
  setTheme = () => {},
} = {}) => ({
  key,
  setTheme,
  themes,
  theme: themes[key],
  themesList: Object.entries(themes) as ThemesList,
});

const ThemeProvider = ({ children }: ThemeProviderProps) => {
  const [key, setKey] = useState(defaultKey);

  const value: ThemeProviderValue = useMemo(
    () =>
      getDefaultValue({
        key,
        setTheme: setKey,
      }),
    [key]
  );

  return (
    <Ctx.Provider value={value}>
      <StyledThemeProvider theme={value.theme}>
        <GlobalStyle />
        {children}
      </StyledThemeProvider>
    </Ctx.Provider>
  );
};

const useThemeProvider = (): ThemeProviderValue => {
  const ctx = useContext(Ctx);

  return ctx ? ctx : getDefaultValue();
};

export { ThemeProvider, useThemeProvider };
