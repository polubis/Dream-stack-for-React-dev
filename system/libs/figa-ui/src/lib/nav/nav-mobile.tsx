import styled from 'styled-components';
import type { NavMobileProps } from './defs';
import { Button } from '../button';
import { CloseIcon } from '../icon';
import { column, row, size, streched } from '../shared';
import { tokens } from '../theme-provider';
import { usePortal, useScrollHide } from '@system/figa-hooks';

const Container = styled.div`
  ${streched('fixed')}
  ${column()}
  background: ${(props) => props.theme.nav.bg};
  z-index: ${tokens.z[400]};
  height: 100vh;

  .nav-mobile-header {
    ${row()}
    flex-shrink: 0;
    height: ${tokens.spacing[1000]};
    border-bottom: ${tokens.spacing[25]} solid
      ${(props) => props.theme.nav.borderColor};

    .nav-mobile-close {
      margin: 0 ${tokens.spacing[250]} 0 auto;
    }
  }

  .nav-mobile-actions {
    ${row()}
    justify-content: flex-end;
    flex-shrink: 0;
    border-top: ${tokens.spacing[25]} solid
      ${(props) => props.theme.nav.borderColor};
    height: ${tokens.spacing[1000]};
    padding: 0 ${tokens.spacing[250]};

    & > *:not(:last-child) {
      margin-right: ${tokens.spacing[200]};
    }
  }

  .nav-mobile-links-wrapper {
    display: flex;
    height: 100%;
    overflow-y: auto;
  }

  .nav-mobile-links {
    ${column()}
    padding: ${tokens.spacing[250]};
    margin: auto;

    & > *:not(:first-child) {
      margin: ${tokens.spacing[300]} 0 0 0;
      text-align: center;
    }

    .divider {
      margin: ${tokens.spacing[450]} auto ${tokens.spacing[150]} auto;

      & > * {
        ${size(tokens.spacing[25], tokens.spacing[250])}
      }
    }
  }
`;

const NavMobile = ({ actions, children, onClose }: NavMobileProps) => {
  useScrollHide();
  const { render } = usePortal();

  return render(
    <Container className="nav-mobile">
      <header className="nav-mobile-header">
        <Button
          className="nav-mobile-close"
          size={2}
          title="Close"
          shape="rounded"
          onClick={onClose}
        >
          <CloseIcon />
        </Button>
      </header>
      <div className="nav-mobile-links-wrapper">
        <ul className="nav-mobile-links">{children}</ul>
      </div>
      <footer className="nav-mobile-actions">{actions}</footer>
    </Container>
  );
};

export { NavMobile };
