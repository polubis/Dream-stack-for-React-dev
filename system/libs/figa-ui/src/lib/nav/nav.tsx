import styled from 'styled-components';
import type { NavProps } from './defs';
import c from 'classnames';
import { M_DOWN, T_DOWN, tokens } from '../theme-provider';
import { row } from '../shared';
import { Button } from '../button';
import { HamburgerIcon } from '../icon';
import { useToggle } from '@system/figa-hooks';
import { Link, type LinkProps } from '../link';
import { NavMobile } from './nav-mobile';
import { Divider } from '../divider';

const Container = styled.nav`
  display: grid;
  width: 100%;
  align-items: center;
  gap: ${tokens.spacing[500]};
  grid-template-columns: 1fr 3fr 1fr;
  height: ${tokens.spacing[1000]};

  .nav-mobile {
    display: none;
  }

  .nav-links-wrapper {
    ${row()}
    width: 100%;
    overflow-x: auto;

    .nav-links {
      ${row()}
      padding: ${tokens.spacing[250]} ${tokens.spacing[200]};
      margin: 0 auto;

      & > *:not(:first-child) {
        margin: 0 0 0 ${tokens.spacing[300]};
      }

      .divider {
        margin: ${0} ${tokens.spacing[150]} 0 ${tokens.spacing[450]};
      }
    }
  }

  .nav-actions {
    ${row()}
    justify-self: end;

    & > *:not(:first-child) {
      margin: 0 0 0 ${tokens.spacing[150]};
    }
  }

  @media ${T_DOWN} {
    grid-template-columns: 1fr auto auto;
    gap: ${tokens.spacing[150]};

    .nav-links-wrapper {
      display: none;
    }

    .nav-actions {
      justify-self: center;
    }

    .nav-mobile {
      display: block;
    }
  }

  @media ${M_DOWN} {
    grid-template-columns: 1fr auto auto;

    .nav-links-wrapper {
      display: none;
    }

    .nav-mobile {
      justify-self: flex-end;
    }
  }
`;

const Nav = ({ className, children, actions, logo }: NavProps) => {
  const toggler = useToggle();

  return (
    <>
      {toggler.opened && (
        <NavMobile actions={actions} onClose={toggler.close}>
          {children}
        </NavMobile>
      )}
      <Container className={c('nav', className)}>
        <div className="nav-logo">{logo}</div>
        <div className="nav-links-wrapper">
          <ul className="nav-links">{children}</ul>
        </div>
        <div className="nav-actions">{actions}</div>
        <div className="nav-mobile">
          <Button
            className="nav-mobile-trigger"
            size={2}
            title="Open"
            shape="rounded"
            onClick={toggler.open}
          >
            <HamburgerIcon />
          </Button>
        </div>
      </Container>
    </>
  );
};

Nav.Divider = () => <Divider axis="y" />;
Nav.Link = (props: LinkProps) => (
  <li>
    <Link {...props} />
  </li>
);

export { Nav };
