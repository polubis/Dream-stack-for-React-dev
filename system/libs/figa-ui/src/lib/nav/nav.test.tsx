import { render } from '@testing-library/react';
import { Nav } from './nav';
import { ThemeProvider } from '../theme-provider';
import { usePortal, useToggle } from '@system/figa-hooks';
import type { ReactNode } from 'react';

jest.mock('@system/figa-hooks');

describe('Nav can be used when: ', () => {
  beforeAll(() => {
    (usePortal as jest.Mock).mockImplementation(() => ({
      render: (children: ReactNode) => children,
    }));
    (useToggle as jest.Mock).mockImplementation(
      jest.requireActual('@system/figa-hooks')['useToggle']
    );
  });

  it('[FRAGILE] assigns classes', () => {
    const { asFragment } = render(
      <ThemeProvider>
        <Nav
          className="my-nav"
          logo={<span>Logo</span>}
          actions={
            <>
              <button>Register</button>
              <button>Sign In</button>
            </>
          }
        >
          <a href="/test">Articles</a>
          <a href="/">Authors</a>
          <a href="/">Admin</a>
        </Nav>
      </ThemeProvider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
