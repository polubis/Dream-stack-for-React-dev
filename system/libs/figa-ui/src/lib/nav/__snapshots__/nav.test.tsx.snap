// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav can be used when:  [FRAGILE] assigns classes 1`] = `
<DocumentFragment>
  <nav
    class="sc-gswNZR cxrvnc nav my-nav"
  >
    <div
      class="nav-logo"
    >
      <span>
        Logo
      </span>
    </div>
    <div
      class="nav-links-wrapper"
    >
      <ul
        class="nav-links"
      >
        <a
          href="/test"
        >
          Articles
        </a>
        <a
          href="/"
        >
          Authors
        </a>
        <div
          class="box divider default y column empty"
        >
          <div />
        </div>
        <a
          href="/"
        >
          Admin
        </a>
      </ul>
    </div>
    <div
      class="nav-actions"
    >
      <button>
        Register
      </button>
      <button>
        Sign In
      </button>
    </div>
    <div
      class="nav-mobile"
    >
      <button
        class="button size-2 rounded filled primary nav-mobile-trigger"
        shape="rounded"
        size="2"
        title="Open"
      >
        <svg
          class="icon hamburger-icon"
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4 7H7M20 7H11M20 17H17M4 17H13M4 12H20"
            stroke="black"
            stroke-linecap="round"
            stroke-width="1.5"
          />
        </svg>
      </button>
    </div>
  </nav>
</DocumentFragment>
`;

exports[`Nav can be used when:  [FRAGILE] open/close mobile navigation 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe kIUTKk nav-mobile"
  >
    <header
      class="nav-mobile-header"
    >
      <button
        class="button size-2 rounded filled primary nav-mobile-close"
        shape="rounded"
        size="2"
        title="Close"
      >
        <svg
          class="icon close-icon"
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6.4 19L5 17.6L10.6 12L5 6.4L6.4 5L12 10.6L17.6 5L19 6.4L13.4 12L19 17.6L17.6 19L12 13.4L6.4 19Z"
          />
        </svg>
      </button>
    </header>
    <div
      class="nav-mobile-links-wrapper"
    >
      <ul
        class="nav-mobile-links"
      >
        <a
          href="/test"
        >
          Articles
        </a>
        <a
          href="/"
        >
          Authors
        </a>
        <div
          class="box divider default y column empty"
        >
          <div />
        </div>
        <a
          href="/"
        >
          Admin
        </a>
      </ul>
    </div>
    <footer
      class="nav-mobile-actions"
    >
      <button>
        Register
      </button>
      <button>
        Sign In
      </button>
    </footer>
  </div>
  <nav
    class="sc-gswNZR cxrvnc nav my-nav"
  >
    <div
      class="nav-logo"
    >
      <span>
        Logo
      </span>
    </div>
    <div
      class="nav-links-wrapper"
    >
      <ul
        class="nav-links"
      >
        <a
          href="/test"
        >
          Articles
        </a>
        <a
          href="/"
        >
          Authors
        </a>
        <div
          class="box divider default y column empty"
        >
          <div />
        </div>
        <a
          href="/"
        >
          Admin
        </a>
      </ul>
    </div>
    <div
      class="nav-actions"
    >
      <button>
        Register
      </button>
      <button>
        Sign In
      </button>
    </div>
    <div
      class="nav-mobile"
    >
      <button
        class="button size-2 rounded filled primary nav-mobile-trigger"
        shape="rounded"
        size="2"
        title="Open"
      >
        <svg
          class="icon hamburger-icon"
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4 7H7M20 7H11M20 17H17M4 17H13M4 12H20"
            stroke="black"
            stroke-linecap="round"
            stroke-width="1.5"
          />
        </svg>
      </button>
    </div>
  </nav>
</DocumentFragment>
`;

exports[`Nav can be used when:  [FRAGILE] open/close mobile navigation 2`] = `
<DocumentFragment>
  <nav
    class="sc-gswNZR cxrvnc nav my-nav"
  >
    <div
      class="nav-logo"
    >
      <span>
        Logo
      </span>
    </div>
    <div
      class="nav-links-wrapper"
    >
      <ul
        class="nav-links"
      >
        <a
          href="/test"
        >
          Articles
        </a>
        <a
          href="/"
        >
          Authors
        </a>
        <div
          class="box divider default y column empty"
        >
          <div />
        </div>
        <a
          href="/"
        >
          Admin
        </a>
      </ul>
    </div>
    <div
      class="nav-actions"
    >
      <button>
        Register
      </button>
      <button>
        Sign In
      </button>
    </div>
    <div
      class="nav-mobile"
    >
      <button
        class="button size-2 rounded filled primary nav-mobile-trigger"
        shape="rounded"
        size="2"
        title="Open"
      >
        <svg
          class="icon hamburger-icon"
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4 7H7M20 7H11M20 17H17M4 17H13M4 12H20"
            stroke="black"
            stroke-linecap="round"
            stroke-width="1.5"
          />
        </svg>
      </button>
    </div>
  </nav>
</DocumentFragment>
`;
